### DISCUSSION TITLE
How to get the final value after dragging the Slider?

### DISCUSSION THREAD

**Discussion by albertbronsky** [mantinedev/mantine#738] [Q&A] [answered]
Dragging the `Slider` triggers lots of unneccessary (intermediate) updates in the state and makes the experience very laggy. Is there a way to get the final value only after sliding is complete?

**Comment by rtivital** [comment, marked as answer]
You can debounce state changes on your side, e.g. using throttle/debouce functions of use-debounced-value hook

**Reply by albertbronsky** [reply]
> You can debounce state changes on your side, e.g. using throttle/debouce functions of use-debounced-value hook

But use-debounced-value hook still requires value in the slider to update every time? How is it different then?

**Reply by nvegater** [reply]
Internally the state is being intermediately updated, but the lagginess is gone because you're not accessing the blinking value of the state, you're using the debounced value as a proxy. (thats my theory as least).

But I agree that ideally there should be a "onClick Release" event to only get the final value to avoid internal overhead. If I wasnt so busy with other 2 projects (one of them using mantine) I'd love to do a PR about this. (If I am not completely wrong).

**Reply by rtivital** [reply]
@nvegater yep, we can add `onChangeEnd` prop support to Slider and RangeSlider components, feel free to contibute this feature. `use-move` hook already supports `onScrubEnd` callback, so that should not be hard. Do not forget about arrow keys events.
